### DISCUSSION TITLE
Use Arquero for query processing and transformation of data

### DISCUSSION THREAD

**Discussion by thinkh** [lineupjs/lineupjs#529] [Ideas]
Recently, I stumbled upon the [Arquero](https://uwdata.github.io/arquero/) library from the Interactive Data Lab from the University of Washington.

I read through parts of the documentation and tried some interactive examples from [their Observable notebooks](https://observablehq.com/@uwdata/arquero?collection=@uwdata/arquero). As far as I can see, it would support all data operations that LineUp requires (and provide even more features):

* Load data from an [array of objects](https://uwdata.github.io/arquero/api/#from), [JSON](https://uwdata.github.io/arquero/api/#fromJSON), [CSV](https://uwdata.github.io/arquero/api/#fromJSON), ...
* [Sort by a single or multiple criteria](https://uwdata.github.io/arquero/api/verbs#orderby)
* [Group by a single or multiple criteria](https://uwdata.github.io/arquero/api/verbs#groupby)
* [Filter by criteria](https://uwdata.github.io/arquero/api/verbs#filter)

I could imagine that we could replace a lot of data operations from the local data provider with Arquero and hence reduce the complexity of our source code. 

@sgratzl What do you think about this?

**Further resources**

* [Observable Notebooks](https://observablehq.com/collection/@uwdata/arquero) with some guides and tutorials
* [Illustrated guide to Arquero verbs](https://observablehq.com/@uwdata/an-illustrated-guide-to-arquero-verbs?collection=@uwdata/arquero)
* [Visual data wrangler](https://observablehq.com/@observablehq/data-wrangler)

**Comment by sgratzl** [comment]
yeah I've heard about it. One thing I wonder regarding lineup, how the notion of `accessors` will complicate things. So far LineUp itself doesn't require to have a plain data structure as input, e.g. the lazy loaded columns are not in the main data but kinda "injected" using a custom accessor.
